### ISSUE TITLE
Multiple TVDB episodes mapped to the same anidb episode

### ISSUE THREAD

**Issue by KurtzPT** [ZeroQI/Hama.bundle#500]
So I found an issue that probably is happening to a bunch of other shows where episodes are being mapped to two different seasons at the same time on TVDB.

Example:
- [AniDB Link](https://anidb.net/anime/13493)
- [TheTVDB Link (Season 3)](https://thetvdb.com/series/sword-art-online/seasons/official/3) 

File to test it:
[!] file:       "/media/Ani-Test/Sword Art Online Alicization (2018) [anidb-13493]/Sword Art Online Alicization  - 03.mkv"

Some logs:

> --- AniDBTVDBMap --------------------------------------------------------------------------------------------------------------------------------------------
[+] AniDBid: 13493, TVDBid: 259640, defaulttvdbseason:    3, offset:   0, TMDBid:        , IMDBid:           , name: Sword Art Online: Alicization
    - season: [ 0],           [ 0], range:       [000-000], offset:   0, text: 1-24;2-25
             -----          ------
             13493          259640
[ ] collection: TVDBid '259640' is part of collection: '['Sword Art Online Collection']', related_anime_list: ['8692', '10022', '10376', '11681', '13493', '14796', '15146']
[ ] studio: 
--- return --------------------------------------------------------------------------------------------------------------------------------------------------
AniDB_id: '13493', AniDB_id2: '13493', AniDBid: '13493', TVDB_id: '', TVDB_id2: '259640', TVDBid: '259640'
mappingList: {
  'defaulttvdbseason': '3',
  'defaulttvdbseason_a': False,
  'episodeoffset': '0',
  'name': 'Sword Art Online: Alicization',
  'possible_anidb3': False,
  's1e1_mapped': False,
  'season_map': {'13493': {'max': '3', 'min': '3'}},
  'TVDB': {'s0e24': ('0', '1', '13493'), 's0e25': ('0', '2', '13493'), 's3': {'13493': '0'}},
  'tvdbcount': 1}

Episodes match:

> [X]    s1e3 s01e003 anidbid: xxxxxxx air_date: 2012-07-22 abs_number: 3, title: The Red-Nosed Reindeer
and

> [X]    s3e3 s01e003 anidbid:   13493 air_date: 2018-10-21 abs_number: 53, title: The End Mountains

---

ASS Logs:

> =============================================================================================================================================================
Call: "Plex", path: "Sword Art Online Alicization (2018) [anidb-13493]", folder_show: "Sword Art Online Alicization (2018) [anidb-13493]", dirs (0), files (1)
=============================================================================================================================================================
Forced ID (series folder) - source: "anidb", id: "13493"
"Sword Art Online Alicization (2018) [anidb-13493]" s01e         3                         "Word Search" "" "Sword Art Online Alicization  - 03.mkv"
=============================================================================================================================================================

This is an issue because in this example it sometimes sets thumbs from the season 1 episode as the default thumb (while downloading the correct season 3 episode thumb).

**Comment by ZyFinity** [comment]
To add to this, I have had the same issue alongside the main posters for the series all being the same when using sorting by anidb.
e.g. Folder structure:
Sword Art Online [anidb-8692]
Sword Art Online II [anidb-10376]
Sword Art Online: Alicization [anidb-13493]

All three of these will grab the same TVDB poster for the show and set it as the main poster, however after looking at the downloaded posters, it seems the correct posters are being downloaded, but just not set as the main poster.
This seems to be a related issue with the thumbnails where "Sword Art Online: Alicization [anidb-13493]" is using thumbnails from the show "Sword Art Online [anidb-8692]" even though it downloads the correct thumbnails but just doesn't seem to default it.

**Comment by ZeroQI** [comment]
Issue is tvdb to abide. Episode mapping as both tvdb épisode point to the same episode in the end:
[X] s1e3 s01e003 anidbid: xxxxxxx air_date: 2012-07-22 abs_number: 3, title: The Red-Nosed Reindeer
and
[X] s3e3 s01e003 anidbid: 13493 air_date: 2018-10-21 abs_number: 53, title: The End Mountains

@ZiFinity not same issue. Check in hama logs at the end the series poster selected and it's priority but pretty sure AniDB is not first in metqdqtq source order in settings be side the quality is often crap and TheTVDB poster can't differentiate series at that level...

**Comment by ZyFinity** [comment]
Oh, can't hama differentiate by grabbing the poster under the anidb show to tvdb season mapping as tvdb seasons have posters? I thought hama was supposed to do this?

**Comment by KurtzPT** [comment]
> Issue is tvdb to abide. Episode mapping as both tvdb épisode point to the same episode in the end: [X] s1e3 s01e003 anidbid: xxxxxxx air_date: 2012-07-22 abs_number: 3, title: The Red-Nosed Reindeer and [X] s3e3 s01e003 anidbid: 13493 air_date: 2018-10-21 abs_number: 53, title: The End Mountains


Mapping from anidb to tvdb comes from the anime-list file and if you check that file mapping what you are saying is not correct:

>   <anime anidbid="13493" tvdbid="259640" defaulttvdbseason="3" episodeoffset="" tmdbid="" imdbid="">
    <name>Sword Art Online: Alicization</name>
    <mapping-list>
      <mapping anidbseason="0" tvdbseason="0">;1-24;2-25;</mapping>
    </mapping-list>
  </anime>

As you can see above, defaulttvdbseason is set to 3 so any episode number 1 mapped to anidb-13493 should always go to s03e01 and NEVER s01e01.

**Comment by BrutuZ** [comment]
@KurtzPT Wait, why would it even convert to Season 3? Unless I'm missing something, `anidb` is meant to keep AniDB's mapping, which is only 1 "season" for regular episodes. If you want to convert to TVDB seasons `anidb2` is the go-to (per https://github.com/ZeroQI/Absolute-Series-Scanner#forcing-the-movieseries-id - SAO is even the used example)

**Comment by KurtzPT** [comment]
> @KurtzPT Wait, why would it even convert to Season 3? Unless I'm missing something, `anidb` is meant to keep AniDB's mapping, which is only 1 "season" for regular episodes. If you want to convert to TVDB seasons `anidb2` is the go-to (per https://github.com/ZeroQI/Absolute-Series-Scanner#forcing-the-movieseries-id - SAO is even the used example)

You didn't understand my point, when I mentioned season 3 I wasn't referring the season in Plex but instead I was referring where hama should grab the Metadata on tvdb. 

So if I use anidb mode, everything on Plex is season 1, but the Metadata is grabbed from anidb and the associated episodes on tvdb (which in this case they are on season 3 in the tvdb website)

But it's clear there's a bug in the agent because it maps all episodes to two different seasons on tvdb, when the mapping file only defines season 3 as the correct mapping.

**Comment by ZyFinity** [comment]
To add, this isn't just with SAO, It's happening with other shows that are also on the mapping file.

**Comment by ZeroQI** [comment]
> To add, this isn't just with SAO, It's happening with other shows that are also on the mapping file.

Hence the 'another mapping' strarting the bug report title...
Duplicate of another issue https://github.com/ZeroQI/Hama.bundle/issues/496

**Comment by KurtzPT** [comment]
> > To add, this isn't just with SAO, It's happening with other shows that are also on the mapping file.
> 
> Hence the 'another mapping' starting the bug report title... Bug report is a duplicate of another issue #496
> 
> The function that maps TheTVDB to AniDB ep number has flaws
> 
> * To be fixed. Bug located in this function: https://github.com/ZeroQI/Hama.bundle/blob/master/Contents/Code/AnimeLists.py#L299
> * failsafe to not allow multiple mappings to the same ep. Failsafe to be implimented after this function call: https://github.com/ZeroQI/Hama.bundle/blob/master/Contents/Code/TheTVDBv2.py#L171

I found that issue before, but since the title of that issue says "same season" I wasn't sure if it was the proper place, since my example is basically episodes mapped to multiple seasons.

**Comment by ZeroQI** [comment]
Different thetvdb episodes mapped to same episode, shouldn't happen and same function is at fault

I haven't been using plex for years, NAS is shutdown and in Ireland, am in France with work laptop, will not reproduce issues. I can try coding without tests like i usually do, here is my attempt at a failsafe to not allow multiple mappings to the same ep. 

Failsafe to be implemented after this function call, should remove bad mappings n the meantime and limit bad metadata: https://github.com/ZeroQI/Hama.bundle/blob/master/Contents/Code/TheTVDBv2.py#L171:

```
        if Dict(mappingList, 'defaulttvdbseason_a'):  season, episode          = '1', str(abs_number)
        else:
               season_, episode_, anidbid_ = AnimeLists.anidb_ep(mappingList, season, episode)
               if Dict(TheTVDB_dict, 'seasons', season_, 'episodes', episode_):  season, episode, anidbid = '0',      '0',      ''
               else:                                                             season, episode, anidbid = season_', episode_, anidbid_
```
or maybe simpler and better: :
```
        if Dict(mappingList, 'defaulttvdbseason_a'):  season, episode          = '1', str(abs_number)
        else:
               season, episode, anidbid = AnimeLists.anidb_ep(mappingList, season, episode)
               if Dict(TheTVDB_dict, 'seasons', season, 'episodes', episode):  continue  #ep and season tuple already mapped, we skip as a failsafe to prevent bad metadata 
```

If somebody can test (KurtsPT?) and see if the mapping to the same valid ep is resolved, then i need to fix the initial issue but need logs to see the mapping before/after and figure out the actual fix.

**Comment by ZeroQI** [comment]
- would be good to enable debug and give me the text output of anidb_ep(mappingList, season, episode) for the correct and impacted episodes, with that i can manage without reproducing and hopefully fix

**Comment by KurtzPT** [comment]
Alright tried out the 1st version:

![image](https://user-images.githubusercontent.com/14930057/152869320-931b670d-1ba7-49e3-a793-44227b98bf85.png)

It didn't work. It matches everything to season 1 ONLY.

```
--- AniDBTVDBMap --------------------------------------------------------------------------------------------------------------------------------------------
[+] AniDBid: 13493, TVDBid: 259640, defaulttvdbseason:    3, offset:   0, TMDBid:        , IMDBid:           , name: Sword Art Online: Alicization
    - season: [ 0],           [ 0], range:       [000-000], offset:   0, text: 1-24;2-25
             -----          ------
             13493          259640
[ ] collection: TVDBid '259640' is part of collection: '['Sword Art Online Collection']', related_anime_list: ['8692', '10022', '10376', '11681', '13493', '14796', '15146']
[ ] studio: 
--- return --------------------------------------------------------------------------------------------------------------------------------------------------
AniDB_id: '13493', AniDB_id2: '13493', AniDBid: '13493', TVDB_id: '', TVDB_id2: '259640', TVDBid: '259640'
mappingList: {
  'defaulttvdbseason': '3',
  'defaulttvdbseason_a': False,
  'episodeoffset': '0',
  'name': 'Sword Art Online: Alicization',
  'possible_anidb3': False,
  's1e1_mapped': False,
  'season_map': {'13493': {'max': '3', 'min': '3'}},
  'TVDB': {'s0e24': ('0', '1', '13493'), 's0e25': ('0', '2', '13493'), 's3': {'13493': '0'}},
  'tvdbcount': 1}
AnimeLists_dict: {
  'collections': ['Sword Art Online Collection']}
=== tvdb4.GetMetadata() =====================================================================================================================================
not tvdb4 mode
=== TheTVDB.GetMetadata() ===================================================================================================================================
TVDBid: '259640', IMDbid: '', language_series : ['jap', 'en'], language_episodes: ['en', 'jap']
--- series --------------------------------------------------------------------------------------------------------------------------------------------------
common.LoadFile() - File cached locally - Filename: 'TheTVDB/json/259640/series_jap.json', Age: '13.88 days', Limit: '6 days', url: 'https://api.thetvdb.com/series/259640?jap'
common.SaveFile() - CachePath: '/config/Library/Application Support/Plex Media Server/Plug-in Support/Data/com.plexapp.agents.hama/DataItems', file: 'TheTVDB/json/259640/series_jap.json'
[ ] language_rank: 2
[ ] title: Sword Art Online
[ ] original_title: Sword Art Online
common.LoadFile() - File cached locally - Filename: 'TheTVDB/json/259640/series_en.json', Age: '13.88 days', Limit: '6 days', url: 'https://api.thetvdb.com/series/259640?en'
common.SaveFile() - CachePath: '/config/Library/Application Support/Plex Media Server/Plug-in Support/Data/com.plexapp.agents.hama/DataItems', file: 'TheTVDB/json/259640/series_en.json'
[ ] language_rank: 2
[ ] title: Sword Art Online
[ ] original_title: Sword Art Online
[ ] IMDbid: tt2250192
[ ] zap2itId: 
[ ] content_rating: TV-14
[ ] originally_available_at: 2012-07-08
[ ] studio: Tokyo MX
[ ] rating: 7.9
[ ] status: Ended
[ ] genres: ['Action', 'Adventure', 'Animation', 'Anime', 'Fantasy', 'Romance', 'Science Fiction', 'Thriller']
[ ] duration: 1500000
--- actors --------------------------------------------------------------------------------------------------------------------------------------------------
common.LoadFile() - File cached locally - Filename: 'TheTVDB/json/259640/actors_en.json', Age: '13.88 days', Limit: '6 days', url: 'https://api.thetvdb.com/series/259640/actors'
common.SaveFile() - CachePath: '/config/Library/Application Support/Plex Media Server/Plug-in Support/Data/com.plexapp.agents.hama/DataItems', file: 'TheTVDB/json/259640/actors_en.json'
[ ] role: Silica                                            , name: Rina Hidaka         , photo: 
[ ] role: Kirigaya Kazuto                                   , name: Yoshitsugu Matsuoka , photo: 
[ ] role: Heathcliff                                        , name: Tooru Ookawa        , photo: 
[ ] role: Ryoutarou "Klein" Tsuboi                          , name: Hiroaki Hirata      , photo: 
[ ] role: Rika "Lisbeth" Shinozaki                          , name: Ayahi Takagaki      , photo: 
[ ] role: Administrator                                     , name: Maaya Sakamoto      , photo: 
[ ] role: Andrew Gilbert "Agil" Mills                       , name: Hiroki Yasumoto     , photo: 
[ ] role: Shinkawa Kyouji                                   , name: Natsuki Hanae       , photo: 
[ ] role: Konno Yuuki                                       , name: Aoi Yuuki           , photo: 
[ ] role: Shinichi Nagata                                   , name: Murase Ayumu        , photo: 
[ ] role: KIRIKALVO                                         , name: Kouichi Yamadera    , photo: 
[ ] role: Sugō Nobuyuki                                     , name: Takehito Koyasu     , photo: 
[ ] role: Yui                                               , name: Kanae Itou          , photo: 
[ ] role: Eugeo                                             , name: Nobunaga Shimazaki  , photo: 
[ ] role: Kirigaya Suguha                                   , name: Ayana Taketatsu     , photo: 
[ ] role: Alice                                             , name: Ai Kayano           , photo: 
[ ] role: Shino "Sinon" Asada                               , name: Miyuki Sawashiro    , photo: 
[ ] role: Sachi                                             , name: Saori Hayami        , photo: 
[ ] role: Yuuki Asuna                                       , name: Haruka Tomatsu      , photo: 
--- episodes ------------------------------------------------------------------------------------------------------------------------------------------------
common.LoadFile() - File cached locally - Filename: 'TheTVDB/json/259640/episodes_page1_en.json', Age: '13.88 days', Limit: '6 days', url: 'https://api.thetvdb.com/series/259640/episodes?page=1'
common.SaveFile() - CachePath: '/config/Library/Application Support/Plex Media Server/Plug-in Support/Data/com.plexapp.agents.hama/DataItems', file: 'TheTVDB/json/259640/episodes_page1_en.json'
common.LoadFile() - File cached locally - Filename: 'TheTVDB/json/259640/episodes_page2_en.json', Age: '13.88 days', Limit: '6 days', url: 'https://api.thetvdb.com/series/259640/episodes?page=2'
common.SaveFile() - CachePath: '/config/Library/Application Support/Plex Media Server/Plug-in Support/Data/com.plexapp.agents.hama/DataItems', file: 'TheTVDB/json/259640/episodes_page2_en.json'
[ ]    s0e1 s00e001 anidbid: xxxxxxx air_date: 2012-10-24
[ ]    s0e2 s00e002 anidbid: xxxxxxx air_date: 2012-12-03
[ ]    s0e3 s00e003 anidbid: xxxxxxx air_date: 2012-12-26
[ ]    s0e4 s00e004 anidbid: xxxxxxx air_date: 2013-01-23
[ ]    s0e5 s00e005 anidbid: xxxxxxx air_date: 2013-02-27
[ ]    s0e6 s00e006 anidbid: xxxxxxx air_date: 2013-03-27
[ ]    s0e7 s00e007 anidbid: xxxxxxx air_date: 2013-04-24
[ ]    s0e8 s00e008 anidbid: xxxxxxx air_date: 2013-05-22
[ ]    s0e9 s00e009 anidbid: xxxxxxx air_date: 2013-06-26
[ ]   s0e10 s00e010 anidbid: xxxxxxx air_date: 2013-12-31
[ ]   s0e11 s00e011 anidbid: xxxxxxx air_date: 2014-12-23
[ ]   s0e12 s00e012 anidbid: xxxxxxx air_date: 2014-10-11
[ ]   s0e13 s00e013 anidbid: xxxxxxx air_date: 2014-10-22
[ ]   s0e14 s00e014 anidbid: xxxxxxx air_date: 2014-11-26
[ ]   s0e15 s00e015 anidbid: xxxxxxx air_date: 2014-12-24
[ ]   s0e16 s00e016 anidbid: xxxxxxx air_date: 2015-01-28
[ ]   s0e17 s00e017 anidbid: xxxxxxx air_date: 2015-02-25
[ ]   s0e18 s00e018 anidbid: xxxxxxx air_date: 2015-03-25
[ ]   s0e19 s00e019 anidbid: xxxxxxx air_date: 2015-04-22
[ ]   s0e20 s00e020 anidbid: xxxxxxx air_date: 2015-05-27
[ ]   s0e21 s00e021 anidbid: xxxxxxx air_date: 2015-06-24
[ ]   s0e22 s00e022 anidbid: xxxxxxx air_date: 2017-02-18
[ ]   s0e23 s00e023 anidbid: xxxxxxx air_date: 2017-09-27
[ ]   s0e24 s00e001 anidbid:   13493 air_date: 2019-02-17
[ ]   s0e25 s00e002 anidbid:   13493 air_date: 2019-10-06
[ ]   s0e26 s00e026 anidbid: xxxxxxx air_date: 2020-04-19
[ ]   s0e27 s00e027 anidbid: xxxxxxx air_date: 2021-10-30
[X]    s1e1 s01e001 anidbid: xxxxxxx air_date: 2012-07-08 abs_number: 1, title: The World of Swords
```

**Comment by KurtzPT** [comment]
@ZeroQI I set debug on on `Info.plist` file, not sure if you have what you wanted above.

**Comment by KurtzPT** [comment]
Second solution:
![image](https://user-images.githubusercontent.com/14930057/152872259-94307d72-9aab-4042-86b6-7c95c4a85f7f.png)

Still the same as the 1st solution:

```
[?] (#1) season: 1, episode: 1
[?] (#3) s1: {}
[?] (#4) defaulttvdbseason: 3, episodeoffset: 0, s1e1_mapped: False
[?] (#5) defaulttvdbseason: 3, episodeoffset: 0, s1e1_mapped: False
[X]    s1e1 s01e001 anidbid: xxxxxxx air_date: 2012-07-08 abs_number: 1, title: The World of Swords
 - [1] title:   [en] The World of Swords
[?] (#1) season: 1, episode: 2
[?] (#3) s1: {}
[?] (#4) defaulttvdbseason: 3, episodeoffset: 0, s1e1_mapped: False
[?] (#5) defaulttvdbseason: 3, episodeoffset: 0, s1e1_mapped: False
[X]    s1e2 s01e002 anidbid: xxxxxxx air_date: 2012-07-15 abs_number: 2, title: Beater
 - [1] title:   [en] Beater
[?] (#1) season: 1, episode: 3
[?] (#3) s1: {}
[?] (#4) defaulttvdbseason: 3, episodeoffset: 0, s1e1_mapped: False
[?] (#5) defaulttvdbseason: 3, episodeoffset: 0, s1e1_mapped: False
[X]    s1e3 s01e003 anidbid: xxxxxxx air_date: 2012-07-22 abs_number: 3, title: The Red-Nosed Reindeer
 - [1] title:   [en] The Red-Nosed Reindeer
[?] (#1) season: 1, episode: 4
[?] (#3) s1: {}
[?] (#4) defaulttvdbseason: 3, episodeoffset: 0, s1e1_mapped: False
[?] (#5) defaulttvdbseason: 3, episodeoffset: 0, s1e1_mapped: False
[X]    s1e4 s01e004 anidbid: xxxxxxx air_date: 2012-07-29 abs_number: 4, title: The Black Swordsman
 - [1] title:   [en] The Black Swordsman
```

```
[?] (#1) season: 3, episode: 1
[?] (#3) s3: {'13493': '0'}
[?] (#3) - offset: 0, anidbid: 13493, int(episode.split('-')[0]): 1
[?] (#1) season: 3, episode: 2
[?] (#3) s3: {'13493': '0'}
[?] (#3) - offset: 0, anidbid: 13493, int(episode.split('-')[0]): 2
[?] (#1) season: 3, episode: 3
[?] (#3) s3: {'13493': '0'}
[?] (#3) - offset: 0, anidbid: 13493, int(episode.split('-')[0]): 3
[?] (#1) season: 3, episode: 4
[?] (#3) s3: {'13493': '0'}
[?] (#3) - offset: 0, anidbid: 13493, int(episode.split('-')[0]): 4
```

**Comment by EndOfLine369** [comment]
Possible issue from https://github.com/ZeroQI/Hama.bundle/commit/b475676bcc72fe2a9076b8d86394d1acf7a87bf0 as is the last change to that function. Try reverting these changes and test.

**Comment by KurtzPT** [comment]
I'll try that tomorrow since it's very late now, but if I revert that PR, it will reintroduce the issue that PR fixed. :/

**Comment by ZyFinity** [comment]
Hi, was there any progress regarding this issue? or is the current fix temperamental?

**Comment by plato79** [comment]
Undoing https://github.com/ZeroQI/Hama.bundle/commit/b475676bcc72fe2a9076b8d86394d1acf7a87bf0 fixes a lot of problems with incorrect title names, though #464 should be fixed with another way it seems.

**Comment by KurtzPT** [comment]
If you revert the #464 fix as it is, it will start to not follow the proper logic that was implemented in the anime-list mapping.

In short, you'll fix one issue and create the issue that was happening in #464, because Hama is not applying the correct logic that is being followed by anime-list guys that create and are maintaining the mapping file.

The proper way would be to make both fixes for #500 and #464 without any of them breaking the other.

**Comment by KurtzPT** [comment]
In short, if `mapping-list` is present, any episode that is matched in it will override the default behaviour of direct episode mapping from anidb to tvdb.

```
    <mapping-list>
      <mapping anidbseason="0" tvdbseason="0">;1-1;2-2;3-3;4-4;5-0;</mapping>
    </mapping-list>
```

So in short looking at the example above, anime-list logic will never put direct episode matchings on `mapping-list` like this, because that's already the default behaviour of the `mapping-list` and it's just a waste of space replicating what by default is. 

Only the episode 5 entry would be defined in `mapping-list` since they need to override that mapping because the default there isn't valid (there's no episode 5 in TVDB)

**Comment by KurtzPT** [comment]
On an endnote, looking at the example I gave in the beginning of this issue:

```
 <anime anidbid="13493" tvdbid="259640" defaulttvdbseason="3" episodeoffset="" tmdbid="" imdbid="">
    <name>Sword Art Online: Alicization</name>
    <mapping-list>
      <mapping anidbseason="0" tvdbseason="0">;1-24;2-25;</mapping>
    </mapping-list>
  </anime>
```

The issue I reported above shouldn't even look at `mapping-list` at all, because it's clearly stated in the mapping file every episode (excluding specials) should be matched to tvdb season 3 (`defaulttvdbseason="3"`) and the `mapping-list` here is clearly stating that the only overrides that can be done are just for season 0 (special ANIDB episodes, defined in `anidbseason="0"` and mapped to `tvdbseason="0"`)

So in short the Hama logic is broken and reverting #464 will make it right.

**Comment by zyfinity01** [comment]
Has this issue been looked at?

**Comment by KurtzPT** [comment]
From last months activity on this project I would say this is not being actively developed by @ZeroQI anymore. I only see PR merges from other people.

Unless someone takes their time around the Python code and make a PR, I don't see this being fixed anytime soon.

**Comment by co-sine** [comment]
> From last months activity on this project I would say this is not being actively developed by @ZeroQI anymore. I only see PR merges from other people.
> 
> Unless someone takes their time around the Python code and make a PR, I don't see this being fixed anytime soon.

Very unfortunate :( , this bug is my only issue with Hama, can't get EVA, Ghost in the Shell and Monogatari Series, etc. right.
Hope someone will take it

**Comment by plato79** [comment]
The problem is still there.

For example for movie "Violet Evergarden Gaiden: Eien to Jidou Shuki Ningyou" instead of "Complete Movie" there is an episode: "Episode 1·"I Love You" and Auto Memoir Dolls" which actually belongs to Violet Evergarden TV series.

**Comment by KurtzPT** [comment]
This was wrongly closed as completed, since the PR merged logic is failing in some cases.

Unfortunately like I said above, there's no active development on Hama and a lot of stuff is broken at the moment (not just this issue). The repository owner @ZeroQI seems busy and there's no intention (at least publicly announced) from him that any of the stuff broken/failing will be fixed in the future.
